src/treetensornetworks/projttns/abstractprojttn.jl

```diff
@@ -133,10 +133,10 @@ function position(
 ) where {V}
   # shift position
   P = shift_position(P, pos)
-  # invalidate environments corresponding to internal edges
-  for e in internal_edges(P)
-    unset!(P.environments, e)
-  end
+  # remove internal edges (out of place)
+  ie = internal_edges(P)
+  newenvskeys = filter(!in(ie), keys(P.environments))
+  P = ProjTTN(pos, P.H, getindices(P.environments, newenvskeys))
   # make all environments surrounding new position
   for e in incident_edges(P)
     make_environment!(P, psi, e)
```

Comment:
Seems like we should also make this out-of-place.

Reply:
l. 142?
I also implemented that, but it's not necessary here, since `getindices` returns a new `Dictionary`. If you prefer, I can push the implementation with `make_environment` (out of place).

Reply:
Also we could split this loop off into a separate function `make_environments`.

Reply:
Yeah, I think relying on that copying behavior is a bit low-level, so best to rewrite in terms of an out-of-place function.

Reply:
I implemented both in place and out of place versions. We may want to get rid of the in-place versions or simplify them to make them fully in-place, right now it's a mix.